src/client/dfuse/dfuse_core.c

```diff
@@ -21,9 +21,48 @@
  * portions thereof marked with this legend must also reproduce the markings.
  */
 
+#include <pthread.h>
+
 #include "dfuse_common.h"
 #include "dfuse.h"
 
+static void *
+dfuse_progress_thread(void *arg)
+{
+	struct dfuse_projection_info *fs_handle = arg;
+	int rc;
+	daos_event_t *dev;
+	struct dfuse_event *ev;
+
+	while (1) {
+
+		errno = 0;
+		rc = sem_wait(&fs_handle->dpi_sem);
+		if (rc != 0) {
+			rc = errno;
+			DFUSE_TRA_ERROR(fs_handle,
+					"Error from sem_wait: %d", rc);
+		}
+
+		if (fs_handle->dpi_shutdown)
+			return NULL;
+
+		rc = daos_eq_poll(fs_handle->dpi_eq, 1,
+				DAOS_EQ_WAIT,
```

Comment:
is this supposed to be WAIT (blocking), or NOWAIT?

Reply:
I'd assumed this was spin until message is received.

Reply:
yes this will be a spin till an event completes, but i assumed you didn't want that since you have a while(1) and you just want to poke the engine to make progress.

Reply:
Whilst dfuse is actively waiting for a response then yes, this thread should busy spin.  The semaphore is to allow it to sleep when there is no I/O.